src/focus/dom-utils.ts

```diff
@@ -15,34 +15,47 @@ export function roundRect(rect: HTMLElement | ClientRect): ClientRect {
 }
 
 /**
- * Returns the common ancestor in the DOM of two nodes. From:
- * http://stackoverflow.com/a/7648545
+ * Returns whether the target DOM node is a child of the root.
+ */
+export function isNodeAttached(node: HTMLElement | null, root: HTMLElement | null): boolean {
+  if (!node || !root) {
+    return false;
+  }
+  return root.contains(node);
+}
+
+/**
+ * Returns whether the provided element is visible.
  */
-export function getCommonAncestor(
-  nodeA: HTMLElement | null,
-  nodeB: HTMLElement | null,
-): HTMLElement | null {
-  if (nodeA === null || nodeB === null) {
-    return null;
+export function isVisible(el: HTMLElement | null): boolean {
+  if (!el) {
+    return false;
   }
 
-  const mask = 0x10;
-  while (nodeA != null && nodeA.parentElement) {
-    nodeA = nodeA.parentElement;
-    // tslint:disable-next-line
-    if ((nodeA.compareDocumentPosition(nodeB) & mask) === mask) {
-      return nodeA;
-    }
+  const style = window.getComputedStyle(el);
+  if (style.display === 'none' || style.visibility === 'hidden') {
+    return false;
   }
-  return null;
+  return true;
 }
 
 /**
- * Returns whether the target DOM node is a child of the root.
+ * Returns if the element can receive focus.
  */
-export function isNodeAttached(node: HTMLElement | null, root: HTMLElement | null) {
-  if (!node || !root) {
+export function isFocusable(el: HTMLElement): boolean {
+  if (el === document.activeElement) {
     return false;
   }
-  return root.contains(node);
+  // to prevent navigating to parent container elements with arc-focus-inside
+  if (document.activeElement !== document.body && document.activeElement.contains(el)) {
+    return false;
+  }
+
+  // Dev note: el.tabindex is not consistent across browsers
+  const tabIndex = el.getAttribute('tabIndex');
```

Comment:
I think this was @saadataz's originally (the diff here is just moving the function out). Iirc UWP/edge makes all elements a default 0 instead of -1 tabIndex or something along those lines?

Reply:
Edge doesnt support `el.tabindex` so we had to fallback to using getAttribute

Reply:
@saadataz would it be reasonable to try to get el.tabIndex as a fast path for other browsers (if/when we have apps on other platforms, e.g. Android/Chromium based tv apps) / Edge eventually when they add support?